### PR TITLE
Add link generators to the front page

### PR DESCRIPTION
**Pull request by Feliix42**
Since the project by now supports numerous different hosters and also self-hosted Gitea repositories, I decided to add two link generators to the front page:

Screenshot of the new front page

This should also simplify the addition of features like nestes project paths (#95), at least from an End-User perspective.

I added a new function to generate the forms and some JS to create the links to reduce load on the server for a redirect.

## Caveats

I am not a front-end person! If you see this and feel it's garbage, feel free to point out problems/address them yourself. :)
This has also literally been the first 31 SLoC Javascript in my whole life, so review with caution. 😄

### CODE REVIEW

**Comment by paolobarbolini** [comment]
Minor stylistic nitpick for the JavaScript side: considering you're already using some ES6 features like Template Strings I think it'd make sense to take advantage of the rest of the ES6 features like [`const`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/const)/[`let`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/let) :smiley:

**Comment by Feliix42** [comment]
Thanks for the hint! I'll try and fix that. :)

**Comment by Feliix42** [comment]
I hope that's better now 🙂

File: assets/links.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+    if (hoster === "gitea") {
+        let baseUrl = formRef.elements["baseUrl"].value;
+
+        // verify that the Base URL is not empty
+        if(baseUrl.length == 0) {
+            formRef.elements["baseUrl"].classList.add("is-danger");
+            document.getElementById("baseUrlHelp").classList.add("is-danger");
+            let hostName = formRef.elements["hosterSelect"].value;
+            document.getElementById("baseUrlHelp").innerHTML = `A Base URL is required for Hosting Provider ${hostName}.`
```
Comment (robjtede) [inline assets/links.js:16]:
```suggestion
            document.getElementById("baseUrlHelp").textContent = `A Base URL is required for Hosting Provider ${hostName}.`
```

File: assets/links.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    let crateVer = formRef.elements["crateVersion"].value;
+
+    if (crateVer.length == 0) {
+        console.log("Aight, Imma get da crate");
+        // default to latest version
+        window.location.href = `/crate/${crate}`;
+    } else {
+        console.log("Got a version??");
+        window.location.href = `/crate/${crate}/${crateVer}`;
+    }
```
Comment (robjtede) [inline assets/links.js:40]:
```suggestion
    if (crateVer.length == 0) {
        // default to latest version
        window.location.href = `/crate/${crate}`;
    } else {
        window.location.href = `/crate/${crate}/${crateVer}`;
    }
```

File: assets/links.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+    let hoster = formRef.elements["hosterSelect"].value.toLowerCase();
+    let owner = formRef.elements["owner"].value;
+    let repoName = formRef.elements["repoName"].value;
+
+    if (hoster === "gitea") {
+        let baseUrl = formRef.elements["baseUrl"].value;
+
+        // verify that the Base URL is not empty
+        if(baseUrl.length == 0) {
```
Comment (paolobarbolini) [inline assets/links.js:12]: Same below

```suggestion
        if(baseUrl.length === 0) {
```

File: assets/links.js (hunk 2 above)
Comment (Feliix42) [reply assets/links.js]: Thanks 😂

**Pull request merged**